Ftgp 9173 update fortnox api

mikaelhagerbro opened the pull request:
What I'm trying to do with this PR is to update the API, so this PR is a rebase on top of the head at the master branch in the base repository (i.e the Fortnox GitHub repository).

Unfortunately we have made bug fixes in our repository instead of updating the API to get them, so that have introduced some conflicts that prevents automatically merge. So I guess we need/should force push it?

I don't know if this is the preferred way of working with external forked repositories and doing update like this, so feedback from GitHub expert is most welcome :)

seckin206 commented:
It seems like they provide a nuget package https://www.nuget.org/packages/Fortnox.NET.SDK/

If our fixes are already implemented by them, I suggest to switch to nuget package :)

mikaelhagerbro commented:
@seckin206 It is good to have the source code, especially when they have bugs in the API code (I have found and fixed one regarding the scanned documents already).
I have also made modifications and added support for downloading files into memory streams, instead of just to files on disk.
So I (unfortunately) think it is better to continue to use the source code since the last update of the API, from Fortnox, was 2017-10-11... and I don't think they will fix the issues and add the features anytime soon.

mikaelhagerbro commented:
I will merge this one now since I need it and it won't affect the monitor code since the sub repo points to a commit and does not follow master.
